packages/modals/src/elements/DrawerModal/DrawerModal.tsx

```diff
@@ -0,0 +1,249 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+/* eslint-disable */
+
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  useMemo,
+  useContext,
+  useCallback,
+  createRef,
+  forwardRef,
+  RefAttributes,
+  HTMLAttributes,
+  PropsWithoutRef,
+  ForwardRefExoticComponent
+} from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import mergeRefs from 'react-merge-refs';
+import { ThemeProps, DefaultTheme, ThemeContext } from 'styled-components';
+import { useModal } from '@zendeskgarden/container-modal';
+import { useDocument } from '@zendeskgarden/react-theming';
+import { useFocusVisible } from '@zendeskgarden/container-focusvisible';
+import { ModalsContext } from '../../utils/useModalContext';
+import {
+  StyledDrawerModalBackdrop,
+  StyledDrawerModal,
+  StyledDrawerModalFooter,
+  StyledDrawerModalFooterItem
+} from '../../styled';
+
+import { Header } from './Header';
+import { Body } from './Body';
+import { Close } from './Close';
+
+interface IStaticDrawerModalExport<T, P>
+  extends ForwardRefExoticComponent<PropsWithoutRef<P> & RefAttributes<T>> {
+  Header: typeof Header;
+  Body: typeof Body;
+  Close: typeof Close;
+  Footer: typeof StyledDrawerModalFooter;
+  FooterItem: typeof StyledDrawerModalFooterItem;
+}
+
+export interface IDrawerModalProps {
+  /**
+   * Determines whether the `DrawerModal` is opened or closed
+   */
+  isOpen?: boolean;
+  /**
+   * HTML attributes to spread onto backdrop element
+   */
+  backdropProps?: any;
+  /**
+   * Callback when a close action has been completed.
+   * Can be triggered from the backdrop and Close icon.
+   * @param {Object} event - DOM event that triggered the close action
+   */
+  onClose?: (event: TransitionEvent) => void;
+  /**
+   * The root ID to use for descendants. A unique ID is created if none is provided.
+   **/
+  id?: string;
+  /**
+   * The DOM element where the drawer modal will be rendered to
+   **/
+  appendToNode?: Element;
+  /**
+   * Determines whether to apply keyboard focus on the drawer modal on mount
+   */
+  focusOnMount?: boolean;
+  /**
+   * Determines whether to return keyboard focus to the element that triggered the drawer modal
+   */
+  restoreFocus?: boolean;
+  children?: any;
+}
+
+export const DrawerModal = forwardRef<
+  HTMLDivElement,
+  IDrawerModalProps & ThemeProps<DefaultTheme> & HTMLAttributes<HTMLDivElement>
```

Comment:
With this using the new `ThemeContext` object we can remove `ThemeProps<DefaultTheme>` I think.